pkg/migrations/op_create_constraint_test.go

```diff
@@ -181,6 +256,98 @@ func TestCreateConstraint(t *testing.T) {
 				// Complete is a no-op.
 			},
 		},
+		{
+			name: "create check constraint on multiple columns",
+			migrations: []migrations.Migration{
+				{
+					Name: "01_add_table",
+					Operations: migrations.Operations{
+						&migrations.OpCreateTable{
+							Name: "users",
+							Columns: []migrations.Column{
+								{
+									Name: "id",
+									Type: "serial",
+									Pk:   ptr(true),
+								},
+								{
+									Name:     "name",
+									Type:     "varchar(255)",
+									Nullable: ptr(false),
+								},
+								{
+									Name:     "email",
+									Type:     "varchar(255)",
+									Nullable: ptr(false),
+								},
+							},
+						},
+					},
+				},
+				{
+					Name: "02_create_constraint",
+					Operations: migrations.Operations{
+						&migrations.OpCreateConstraint{
+							Name:    "check_name_email",
+							Table:   "users",
+							Type:    "check",
+							Check:   ptr("name != email"),
+							Columns: []string{"name", "email"},
+							Up: migrations.OpCreateConstraintUp(map[string]string{
+								"name":  "name",
+								"email": "(SELECT CASE WHEN email ~ '@' THEN email ELSE email || '@example.com' END)",
+							}),
+							Down: migrations.OpCreateConstraintDown(map[string]string{
+								"name":  "name",
+								"email": "email",
+							}),
+						},
+					},
+				},
+			},
+			afterStart: func(t *testing.T, db *sql.DB, schema string) {
+				// The new (temporary) column should exist on the underlying table.
+				ColumnMustExist(t, db, schema, "users", migrations.TemporaryName("name"))
+				// The new (temporary) column should exist on the underlying table.
+				ColumnMustExist(t, db, schema, "users", migrations.TemporaryName("email"))
+				// The check constraint exists on the new table.
+				CheckConstraintMustExist(t, db, schema, "users", "check_name_email")
+
+				// Inserting values into the old schema that the check must succeed.
+				MustInsert(t, db, schema, "01_add_table", "users", map[string]string{
+					"name":  "alice",
+					"email": "alice",
+				})
+
+				// Inserting values into the new schema that violate uniqueness should fail.
```

Comment:
This comment should be above line 327 and this comment should read:

```suggestion
				// Inserting values into the new schema that meet the check constraint should succeed.
```